kmod/src/inode.c

```diff
@@ -2015,6 +2065,20 @@ int scoutfs_inode_walk_writeback(struct super_block *sb, bool write)
 	return ret;
 }
 
+bool scoutfs_inode_worm_denied(struct inode *inode)
+{
+	struct scoutfs_inode_info *si = SCOUTFS_I(inode);
+	struct timespec cur_time = CURRENT_TIME;
+	bool worm_set = false;
+
+	if (SCOUTFS_WORM_V1_BIT  == si->worm_bits &&
```

Comment:
This can be the simpler if ((bits & V1) && _compare < 0) return true;  Don't need the worm_set and we don't need to special case tv_sec 0.